File: internal/app/repo/postgres/config_repo.go
package postgres

import (
	"database/sql"

	"go.uber.org/dig"
)

type (
	ConfigRepo interface {
		GetConfigByKey(key string) (string, error)
		UpdateConfigByKey(key string, value string, operatedBy string) error
	}

	ConfigRepoImpl struct {
		dig.In

		*sql.DB
	}
)

func NewConfigRepo(impl ConfigRepoImpl) ConfigRepo {
	return &impl
}

func (c *ConfigRepoImpl) GetConfigByKey(key string) (string, error) {
	var value string
	err := c.QueryRow("SELECT value FROM config WHERE key = $1", key).Scan(&value)
	if err != nil {
		return "", err
	}
	return value, nil
}

func (c *ConfigRepoImpl) UpdateConfigByKey(key string, value string, operatedBy string) error {
	_, err := c.Exec("UPDATE config SET value = $1, updated_at = NOW(), updated_by = $3 WHERE key = $2", value, key, operatedBy)
	if err != nil {
		return err
	}
	return nil
}

File: src/repo/config_repo.ts
import { Pool } from "pg";
import db from "../db/db.config";

export class ConfigRepository {
  private pg: Pool;

  constructor() {
    this.pg = db;
  }

  /**
   * 🧩 GetConfigByKey
   * Sama seperti func (c *ConfigRepoImpl) GetConfigByKey(key string)
   */
  async getConfigByKey(key: string): Promise<string> {
    try {
      const result = await this.pg.query(
        "SELECT value FROM config WHERE key = $1",
        [key]
      );
      if (result.rowCount === 0) {
        throw new Error(`Config not found for key: ${key}`);
      }
      return result.rows[0].value;
    } catch (err: any) {
      console.error(`[repo][config][getConfigByKey] error: ${err.message}`);
      throw err;
    }
  }

  /**
   * 🧩 UpdateConfigByKey
   * Sama seperti func (c *ConfigRepoImpl) UpdateConfigByKey(key, value, operatedBy)
   */
  async updateConfigByKey(
    key: string,
    value: string,
    operatedBy: string
  ): Promise<void> {
    try {
      await this.pg.query(
        "UPDATE config SET value = $1, updated_at = NOW(), updated_by = $3 WHERE key = $2",
        [value, key, operatedBy]
      );
    } catch (err: any) {
      console.error(`[repo][config][updateConfigByKey] error: ${err.message}`);
      throw err;
    }
  }
}
